superset-frontend/src/components/ListView/Filters.tsx

```diff
@@ -119,6 +121,25 @@ function SelectFilter({
         setSelectedOption(matchingOption);
       }
     }
+
+    // Check whether list should be sorted (alphanumerically, ascending)
+    if (sort) {
```

Comment:
We probably want to add the sorting on the backend. In the case that the results are paginated, this sorting would only be on each page and not across the entire result set. We can likely add another entry [here](https://github.com/preset-io/incubator-superset/blob/master/superset-frontend/src/views/CRUD/utils.tsx#L38) for sorting.

Reply:
Updated! I added a default `order_column` for the generic related/distinct sorts, although `datasets` actually has its own custom fetch method.